docs/basics/userguide/seeding.md

```diff
@@ -0,0 +1,376 @@
+    seed=seeder,
+)
+
+first_experiment.fit()
+
+output = evaluate_agents(
+    [first_experiment], n_simulations=5, plot=False
+)  # evaluate the experiment on 5 simulations
+print(output)
+```
+
+Run 1:
+```none
+[INFO] 14:46: Running ExperimentManager fit() for PPO_first_experimentCartPole-v1 with n_fit = 1 and max_workers = None.
+[INFO] 14:46: ... trained!
+[INFO] 14:46: Evaluating PPO_first_experimentCartPole-v1...
+[INFO] Evaluation:.....  Evaluation finished
+   PPO_first_experimentCartPole-v1
+0                             23.3
+1                             19.7
+2                             23.0
+3                             18.8
+4                             19.7
+```
+
+Run 2 :
+```none
+[INFO] 14:46: Running ExperimentManager fit() for PPO_first_experimentCartPole-v1 with n_fit = 1 and max_workers = None.
+[INFO] 14:46: ... trained!
+[INFO] 14:46: Evaluating PPO_first_experimentCartPole-v1...
+[INFO] Evaluation:.....  Evaluation finished
+   PPO_first_experimentCartPole-v1
+0                             23.3
+1                             19.7
+2                             23.0
+3                             18.8
+4                             19.7
+```
+
+
+Run 3 :
+```none
+[INFO] 14:46: Running ExperimentManager fit() for PPO_first_experimentCartPole-v1 with n_fit = 1 and max_workers = None.
+[INFO] 14:46: ... trained!
+[INFO] 14:46: Evaluating PPO_first_experimentCartPole-v1...
+[INFO] Evaluation:.....  Evaluation finished
+   PPO_first_experimentCartPole-v1
+0                             23.3
+1                             19.7
+2                             23.0
+3                             18.8
+4                             19.7
+```
+
+
+</br>
+
+
+**With the seeder, the outputs are the same (reproducible).**
+
+
+</br>
+
+
+
+
+### multi-threading
+If you want use multi-threading, a seeder can spawn other seeders that are independent from it.
+This is useful to seed two different threads, using seeder1 in the first thread, and seeder2 in the second thread.
+```python
+from rlberry.seeding import Seeder
+
+seeder = Seeder(123)
+seeder1, seeder2 = seeder.spawn(2)
+
+print("random sample 1 from seeder1 rng: ", seeder1.rng.normal())
+print("random sample 2 from seeder1 rng: ", seeder1.rng.normal())
+print("-----")
+print("random sample 1 from seeder2 rng: ", seeder2.rng.normal())
+print("random sample 2 from seeder2 rng: ", seeder2.rng.normal())
+```
+```none
+random sample 1 from seeder1 rng:  -0.4732958445958833
+random sample 2 from seeder1 rng:  0.5863995575997462
+-----
+random sample 1 from seeder2 rng:  -0.1722486099076424
+random sample 2 from seeder2 rng:  -0.1930990650226178
+```
+
+</br>
+
+## External libraries
+You can also use a seeder to seed external libraries (such as torch) using the method `set_external_seed`.
+It will be usefull if you want reproducibility with external libraries. In this example, you will use `torch` to generate random numbers.
```

Comment:
maybe also explain how gymnasium is seeded?

Reply:
I don't think it's seeded, there is only torch so far... (except if you use the parameter "seed" when you use a Gymnasium environment)